pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -304,72 +306,81 @@ func (c *SyncController) createDefaultLimitRange(ns *corev1.Namespace, spec *cor
 	return err
 }
 
-func (c *SyncController) validateAndSetNamespaceQuota(ns *corev1.Namespace, quotaToUpdate string) (bool, *corev1.Namespace, error) {
+func (c *SyncController) validateAndSetNamespaceQuota(ns *corev1.Namespace, quotaToUpdate *v32.NamespaceResourceQuota) (bool, *corev1.Namespace, corev1.ResourceList, error) {
 	if ns == nil || ns.DeletionTimestamp != nil {
-		return true, ns, nil
+		return true, ns, nil, nil
 	}
 
 	// get project limit
 	projectLimit, projectID, err := getProjectResourceQuotaLimit(ns, c.ProjectLister)
 	if err != nil {
-		return false, ns, err
+		return false, ns, nil, err
 	}
 
 	if projectLimit == nil {
-		return true, ns, err
+		return true, ns, nil, err
 	}
 
 	updatedNs := ns.DeepCopy()
-	if quotaToUpdate != "" {
+	if quotaToUpdate != nil {
 		if updatedNs.Annotations == nil {
 			updatedNs.Annotations = map[string]string{}
 		}
-		updatedNs.Annotations[resourceQuotaAnnotation] = quotaToUpdate
+		b, err := json.Marshal(quotaToUpdate)
+		if err != nil {
+			return false, nil, nil, err
```

Comment:
Nit: in this case, for the second return value, would we want to return the original namespace rather than nil?

I don't think that it really matter much given that the caller really should check for an error before attempting to use any of the values returned from this, but I figured its at least worth discussing.